### PR TITLE
Allow setting ExternalFile as icon in ORM

### PR DESCRIPTION
**Pull request by FlorianWilhelm**
Allow users to easily set an emoji or external file. Also allows using `:thumps_up:` strings to define emotions. 

Besides that some typos and wrong docstring parameters were fixed.

### CODE REVIEW

File: src/notional/text.py [hunk 1]
```diff
@@ -25,6 +27,16 @@ def markdown(*rtf):
...
 
+def is_emoji(text):
+    """Checks if text is a single emoji"""
+    count = 0
+    for emoji in EMOJI_DATA:
+        count += text.count(emoji)
+        if count > 1:
+            return False
+    return bool(count)
+
```
Comment (jheddings) [inline src/notional/text.py:38]: Would this be more clear?

```python
def is_emoji(text):
     for ch in text:
         if ch not in EMOJI_DATA:
            return False
     return True
```

File: src/notional/orm.py [hunk 2]
```diff
@@ -5,11 +5,14 @@
...
+from typing import Union
 
 from .records import Database, DatabaseRef, Page
 from .schema import PropertyObject, RichText
-from .text import make_safe_python_name
-from .types import EmojiObject, PropertyValue
+from .text import make_safe_python_name, is_emoji
+from .types import EmojiObject, PropertyValue, ExternalFile
+
+from emoji import emojize
```
Comment (jheddings) [inline src/notional/orm.py:15]: Sweet!  I've been meaning to look for such a library...

File: src/notional/orm.py [hunk 3]
```diff
@@ -215,19 +218,27 @@ def icon(self):
...
         """
 
-        if isinstance(emoji, str):
-            emoji = EmojiObject[emoji]
-
-        elif not isinstance(emoji, EmojiObject):
-            raise ValueError("Invalid emoji; unsupported type")
+        if isinstance(icon, str):
+            if icon.startswith(":"):
+                icon = emojize(icon, language='alias')
```
Comment (jheddings) [inline src/notional/orm.py:229]: Prefer double quotes for style consistency...  Black should take care of this for you if you use the `pre-commit` hooks.

File: src/notional/text.py (hunk 1 above)
Comment (FlorianWilhelm) [reply src/notional/text.py:38]: That's right, changed it.

File: src/notional/orm.py (hunk 3 above)
Comment (FlorianWilhelm) [reply src/notional/orm.py]: My bad, forgot to run `pre-commit install` in my new environment. Fixed this.

**Comment by FlorianWilhelm** [comment]
@jheddings, thanks for your review. All points are addressed I guess :-)

**Review by jheddings — APPROVED** [review]
Looks great.  Thanks for the PR!

**Pull request merged**